Framework/PythonInterface/plugins/algorithms/WorkflowAlgorithms/FindPeakAutomatic.py

```diff
@@ -0,0 +1,292 @@
+        self.declareProperty('EndXValue', np.Inf,
+                             doc='Value of X to stop the search to',
+                             validator=FloatBoundedValidator(lower=0.0))
+        self.declareProperty('AcceptanceThreshold', 0.01,
+                             doc='Threshold for considering a peak significant, the exact meaning of the value depends '
+                             'on the cost function used and the data to be fitted. '
+                             'Good values might be about 1-10 for poisson cost and 0.0001-0.01 for chi2',
+                             validator=FloatBoundedValidator(lower=0.0))
+        self.declareProperty('SmoothWindow', 5,
+                             doc='Half size of the window used to find the background values to subtract',
+                             validator=IntBoundedValidator(lower=0))
+        self.declareProperty('BadPeaksToConsider', 20,
+                             doc='Number of peaks that do not exceed the acceptance threshold to be searched before '
+                             'terminating. This is useful because sometimes good peaks can be found after '
+                             'some bad ones. However setting this value too high will make the search much slower.',
+                             validator=IntBoundedValidator(lower=0))
+        self.declareProperty('UsePoissonCost', False,
+                             doc='Use a probabilistic approach to find the cost of a fit instead of using chi2.')
+        self.declareProperty('EstimatePeakSigma', 3.0,
+                             doc='A rough estimate of the standard deviation of the gaussian used to fit the peaks',
+                             validator=FloatBoundedValidator(lower=0.0))
+        self.declareProperty('MinPeakSigma', 0.5,
+                             doc='Minimum value for the standard deviation of a peak',
+                             validator=FloatBoundedValidator(lower=0.0))
+        self.declareProperty('MaxPeakSigma', 30.0,
+                             doc='Maximum value for the standard deviation of a peak',
+                             validator=FloatBoundedValidator(lower=0.0))
+        self.declareProperty('PlotPeaks', True, 'Plot the position of the peaks found by the algorithm')
+
+        # Output table
+        self.declareProperty(name='PeakPropertiesTableName',
+                             defaultValue='',
+                             doc='Name of the table containing the properties of the peaks')
+        self.declareProperty(name='RefitPeakPropertiesTableName',
+                             defaultValue='',
+                             doc='Name of the table containing the properties of the peaks that had to be fitted twice '
+                                 'as the first time the error was unreasonably large')
+
+    def PyExec(self):
+        self.acceptance = float(self.getPropertyValue('AcceptanceThreshold'))
```

Comment:
You can just do `self.acceptance = self.getProperty('AcceptanceThreshold')` rather than grabbing the string and converting it. Similarly for the other properties.